core/templates/pages/contributor-dashboard-page/translation-language-selector/translation-language-selector.component.spec.ts

```diff
@@ -183,6 +188,46 @@ describe('Translation language selector', () => {
     expect(component.activeLanguageCode).toBe('en');
   });
 
+  it('should display the preferred language when the preferred' +
+    ' language is defined', async(() => {
+    component.activeLanguageCode = null;
+    preferredLanguageCode = 'en';
+    // This throws "TS2532". We need to suppress this error because
+    // "constants.SUPPORTED_AUDIO_LANGUAGES" is always defined. Also, it must
+    // contain description for language code 'en' and if it doesn't
+    // then this test should fail.
+    // @ts-ignore
+    const languageDescription = AppConstants.SUPPORTED_AUDIO_LANGUAGES.find(
+      e => e.id === 'en').description;
+
+    spyOn(component.setActiveLanguageCode, 'emit').and.callFake(
+      (languageCode: string) => {
+        component.activeLanguageCode = languageCode;
+      });
+
+    component.ngOnInit();
+
+    fixture.whenStable().then(() => {
+      fixture.detectChanges();
+      expect(component.setActiveLanguageCode.emit)
+        .toHaveBeenCalledWith(preferredLanguageCode);
+      expect(component.activeLanguageCode).toBe(preferredLanguageCode);
+      expect(component.languageSelection).toBe(languageDescription);
```

Comment:
Should we add a pre-check for this before line 208?

Reply:
I have specifically declared `activeLanguageCode` and `languageSelection` in this test.
```
    component.activeLanguageCode = null;
    component.languageSelection = '';
``` 
So now, I think having a pre-check won't be required.